enhancements/olm-deletes-operands.md

```diff
@@ -0,0 +1,209 @@
+
+So to facilitate automatically removing operator managed resources OLM can trigger cleanup via CR removal before the operator is removed.
+
+### Goals
+
+- Clean up includes removal of CRs (operands) and CRDs
+- CRs and CRDs are not automatically removed
+- A user can opt-in to the clean up behavior at operator uninstall time
+
+### Non-Goals
+
+- OLM handles cleanup of anything other than CRs
+- Handle custom cleanup logic like filtering/ordering the deletion of CRs
+
+### Possible Future Goals (Not in scope)
+
+- OLM should fire an alert event if the operator is stalled or taking too long on operand cleanup
+- The timeout for firing an alert event should be user configurable
+- Once [operator conditions][operator-conditions] are supported, there could be an OLM supported condition that the operator can use to indicate operand cleanup failure
+- The `operator-sdk` and `kubectl operator` can use OLM's operand cleanup feature instead of [deleting the CRDs][kb-operator-crd-delete] manually.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator user I want my operator uninstall to also clean up operator-created resources so that I don't have to manually clean up any leftover resources.
+
+### Implementation Details/Notes/Constraints
+
+#### Where to configure cleanup: CSV and/or Subscription
+
+The API to configure operand cleanup could be specified on either the Subscription or ClusterServiceVersion (CSV).
+
+If the cleanup API is only present on the Subscription then operators installed without Subscriptions (by manually creating the CSV and fulfilling requirements) can not opt-in to operand cleanup.
+
+The CSV is the more appropriate point to configure the cleanup option for an operator as the CSV represents the actual installation of the operator, whereas the Subscription only represents the intent to subscribe to updates from the catalog. Moreover, since deleting the CSV causes the operator to be removed, deleting the CSV should be the trigger to initiate operand cleanup. See the [section on CSV finalizers][finalizers-on-csvs] for more details.
+
+#### Deleting CRs
+
+The operand cleanup process, simply put, is the removal of all CRs being reconciled by an operator so it can run finalizers for each CR and clean up any associated operator-created resources.
+
+From OLM's perspective these would be all CRs in the `OperatorGroup`'s [target namespaces][target-namespaces] that are of the same type as the [owned APIservices][owned-apiservices] listed in the CSV. This represents the set of all CRs that the operator reconciles and is responsible for cleaning up after.
+
+OLM should not delete CRs of the type [required APIService][required-apiservices] as those are likely created by the operator, and whose cleanup should be handled by the operator's finalizer.
```

Comment:
Similarly to the above, this is missing [required CRDs](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#required-crds)